src/tools/base64.cpp

```diff
@@ -1,172 +1,115 @@
-    64, 64, 64, 64, 64, 62, 64, 64, 64, 63, 52, 53, 54, 55, 56, 57, 58, 59, 60,
-    61, 64, 64, 64, 64, 64, 64, 64, 0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10,
-    11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 64, 64, 64, 64,
-    64, 64, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42,
-    43, 44, 45, 46, 47, 48, 49, 50, 51, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-    64, 64, 64, 64, 64, 64, 64, 64, 64};
-
-size_t Base64decode_len(const char* bufcoded) {
-  size_t nbytesdecoded;
-  const unsigned char* bufin;
-  int nprbytes;
-
-  bufin = reinterpret_cast<const unsigned char*>(bufcoded);
-  while (pr2six[*(bufin++)] <= 63)
-    ;
-
-  nprbytes = (bufin - (const unsigned char*)bufcoded) - 1;
-  nbytesdecoded = ((nprbytes + 3) / 4) * 3;
-
-  return nbytesdecoded + 1;
-}
-
-int Base64decode(char* bufplain, const char* bufcoded) {
-  int nbytesdecoded;
-  const unsigned char* bufin;
-  unsigned char* bufout;
-  int nprbytes;
-
-  bufin = (const unsigned char*)bufcoded;
-  while (pr2six[*(bufin++)] <= 63)
-    ;
-  nprbytes = (bufin - (const unsigned char*)bufcoded) - 1;
-  nbytesdecoded = ((nprbytes + 3) / 4) * 3;
-
-  bufout = (unsigned char*)bufplain;
-  bufin = (const unsigned char*)bufcoded;
-
-  while (nprbytes > 4) {
-    *(bufout++) = (unsigned char)(pr2six[*bufin] << 2 | pr2six[bufin[1]] >> 4);
-    *(bufout++) =
-        (unsigned char)(pr2six[bufin[1]] << 4 | pr2six[bufin[2]] >> 2);
-    *(bufout++) = (unsigned char)(pr2six[bufin[2]] << 6 | pr2six[bufin[3]]);
-    bufin += 4;
-    nprbytes -= 4;
+#include "base64.h"
+#include "spdlog/spdlog.h"
+
+using namespace cloudevents;
+static const char* B64chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+static const int B64index[128] =
+{
+    0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+    0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+    0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  62, 63, 62, 62, 63,
+    52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 0,  0,  0,  0,  0,  0,
+    0,  0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14,
+    15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 0,  0,  0,  0,  63,
+    0,  26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40,
+    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51
+};
+
+std::string base64::base64encode(const void* data, const size_t &len)
+{
+  if (data == NULL || len == 0 ) {
+      spdlog::error("Invalid data in {}\n", __func__);
```

Comment:
no need to print the function

Reply:
I thought the "__func__" will give exact location of the error.
As per recommendation it is removed.

Reply:
Fixed